core/src/main/java/org/sonarsource/sonarlint/core/SonarLintBackendImpl.java

```diff
@@ -45,79 +52,83 @@ public SonarLintBackendImpl(SonarLintClient client) {
   @Override
   public CompletableFuture<Void> initialize(InitializeParams params) {
     return CompletableFuture.runAsync(() -> {
-      if (initialized.compareAndSet(false, true)) {
-        sonarLintBackend = new InitializedSonarLintBackend(client, params);
+      if (initializeCalled.compareAndSet(false, true) && !initialized.get()) {
```

Comment:
This change is not strictly related to the Spring migration, but is a way to make the initialization sequence safer. Let me know if you prefer to have it in a separate commit.